### DISCUSSION TITLE
Can Variorum be destructive?

### DISCUSSION THREAD

**Discussion by brandonbiggs** [LLNL/variorum#540] [Q&A] [answered]
Hi! 

Thanks for all your work on this tool. I've been reading through your documentation and I was curious, can Variorum be destructive? Can I set power for cpus/gpus too high/low? I don't want to do this of course, but I was curious if it could be done on accident. Maybe if I forgot a 0 or added an extra 0? Do hardware vendors build in power limits that can't be overwritten? Can those limitations be overwritten such that improper use of Variorum is physically destructive?

If this is in the documentation, I apologize for missing it. Also please let me know if there's a better place to ask questions. 

Thanks again!

**Comment by rountree** [comment, marked as answer]
Hi @brandonbiggs,

The short answer is "no," Variorum is not destructive.

That's nothing we can guarantee---Variorum is only a conduit to the API exposed by the operating system (and via that, the processor).  That said, an AMD engineer shared a story with me year about about how a hardware review site tried running a new AMD processor at max load having taken off the heat sink first.  The processor cooked itself, the reviews said AMD processors were garbage, and ever since then every processor contains enough thermal, current, and power sensors to make sure no damage occurs, even when operating outside of required parameters.

The power controls on processors we support have been around for over a decade.  They're solid.  It used to be possible to slow a machine to a crawl with a too-low power bound, but that's easy enough to fix on a reboot.

Variorum does give you an API to change arbitrary MSRs, and if your sysadmin opens up the OS to allow that, it's certainly possible to crash the machine.  Again, nothing a reboot won't fix.

You're good to go.

**Comment by brandonbiggs** [comment]
Thank you @rountree for the great answer! I appreciate the information.

You made a good distinction about Variorum only being the conduit to the exposed APIs. I didn't mean that Variorum would do the damage, but if I could do the damage since Variorum gave me access to those APIs. What you said makes sense though. I'll make sure I can reboot the box before I mess with anything too bad 😄
